scripts/upgrade

```diff
@@ -101,25 +98,63 @@ then
   if 	[ -d "$backup_squelettes" ]
   then
     cp -ar $backup_squelettes $final_path/www/squelettes
-    ynh_secure_remove  $backup_squelettes
-  fi
+   fi
 
   if [ -e "$backup_bdd" ]
   then
 	  cp -a $backup_bdd $final_path/association.sqlite
-    ynh_secure_remove $backup_bdd
   fi
 
   chown -R $app:$app $final_path
+  chmod 755 $final_path
 
-  key=$(ynh_string_random --length=50)
-  cp -a $final_path/config.dist.php $final_path/config.local.php
-  ynh_replace_string --match_string="const SECRET_KEY = '3xUhIgGwuovRKOjVsVPQ5yUMfXUSIOX2GKzcebsz5OINrYC50r';" --replace_string="const SECRET_KEY = '$key';" --target_file="$final_path/config.local.php"
-  if [ "$path_url" == "/" ]; then
-  ynh_replace_string --match_string="//const WWW_URI = '/asso/';" --replace_string="const WWW_URI = '$path_url';" --target_file="$final_path/config.local.php"
-  else
-  ynh_replace_string --match_string="//const WWW_URI = '/asso/';" --replace_string="const WWW_URI = '$path_url/';" --target_file="$final_path/config.local.php"
+  # Create the visitors permission if needed
+   visitors_enabled=$(ynh_permission_has_user "main" "visitors" && echo 0 || echo 1)
+  if [[ $visitors_enabled -eq 0 ]]; then
```

Comment:
Isn't it "-eq 1" here ? I mean apparently 0=yes and 1=no

We only need to temporarily add the visitor perm only if it's not there yet ..?

Reply:
Yes, I haven't really understood these lines, they have been suggested to «temporarily add the visitor perm only if it's not there yet» as you say by @tituspijean in this commit https://github.com/YunoHost-Apps/garradin_ynh/commit/864e1f68cfd0e549b57a457a726f92b2d87b4d99
I am not sure...

Reply:
Looking the app_helpers, I think you are right.  This is here to accept the command curl. So it will be:

```
if [[ $visitors_enabled -eq 1 ]]; then
    ynh_permission_update --permission "main" --add "visitors"
  fi

  ynh_systemd_action --service_name=nginx --action=reload
  # Finalize the upgrade
  ynh_local_curl "/index.php"
  sleep 5
  ynh_local_curl "/index.php"
  upgrade_type="UPGRADE_APP"
```

I am not sure with the next step:

```
  if [[ $visitors_enabled -eq 0 ]]; then
    ynh_permission_update --permission "main" --remove "visitors"
  fi
```
Does it take care of the permission updated just before ?

Reply:
It is  just another condition, when the app has been installed as public, but in this case why remove "visitors" ? Or may be it just a security before the next upgrade ? I have no idea ?

Reply:
Does the first method choosen at the first intallation for "public" or "private" is always mandatory ?

Reply:
I make a test and there is a problem, first may app was public on first install (with branch 1.0.7), then after a successful update, it has only `all_users` and no more 'visitors'...

Reply:
Ok, so I think I understood, twoo conditions, or is public or is private, so it will be:

```
# Create the visitors permission if needed
  visitors_enabled=$(ynh_permission_has_user "main" "visitors" && echo 0 || echo 1)
  if [[ $visitors_enabled -eq 1 ]]; then
    ynh_permission_update --permission "main" --add "visitors"
    ynh_systemd_action --service_name=nginx --action=reload
    # Finalize the upgrade
    ynh_local_curl "/index.php"
    sleep 5
    ynh_local_curl "/index.php"
    upgrade_type="UPGRADE_APP"
    ynh_permission_update --permission "main" --remove "visitors"
  fi

  if [[ $visitors_enabled -eq 0 ]]; then
    ynh_systemd_action --service_name=nginx --action=reload
    # Finalize the upgrade
    ynh_local_curl "/index.php"
    sleep 5
    ynh_local_curl "/index.php"
    upgrade_type="UPGRADE_APP"
  fi
```
do you agree @alexAubin ?

Reply:
I confirm after some tests, like this is seems more logic.

Reply:
resolved in https://github.com/YunoHost-Apps/garradin_ynh/pull/63/commits/0e6e4467cf7525e18b3d5cc558a01d3725216a02